vrrpsyncd/vrrpsync.cpp

```diff
@@ -0,0 +1,640 @@
+        return false;
+    }
+    mac = it->second;
+    SWSS_LOG_INFO("System MAC is %s", mac.c_str());
+    return true;
+}
+
+bool VrrpSync::getAutoIPv6LL(const string& ifname, string& ipv6ll)
+{
+	// std::stringstream cmd;
+	std::string mac;
+	int idx;
+	uint8_t ll_addr[6];
+	struct in6_addr l3_addr;
+    std::size_t pos;
+    string file = "/sys/class/net/"+ifname+"/address";
+    bool exists = false;
+
+    if (ifname == "eth0")
+    {
+        mac = systemMac;
+    }
+    else
+    {
+        mac = ReadLineFromFile(file, &exists);
+        if (!exists)
+        {
+            SWSS_LOG_ERROR("Unable to get MAC on interface %s", ifname.c_str());
+            ipv6ll = "::";
+            return false;
+        } 
+    }
+
+	std::string delimiter = ":";
+	std::string token;
+	idx = 0;
+    while ((pos = mac.find(delimiter)) != std::string::npos) {
+        token = mac.substr(0, pos);
+		ll_addr[idx++] = (uint8_t)std::stoi(token, 0, 16);
+        mac.erase(0, pos + delimiter.length());
+	}
+    token = mac.substr(0, pos);
+	ll_addr[idx] = (uint8_t)std::stoi(token, 0, 16);
+
+	/*Generate LL based interface mac*/
+	l3_addr.s6_addr[0] = 0xfe;
+	l3_addr.s6_addr[1] = 0x80;
+	l3_addr.s6_addr16[1] = 0;
+	l3_addr.s6_addr32[1] = 0;
+	l3_addr.s6_addr[8] = ll_addr[0] | 0x02;
+	l3_addr.s6_addr[9] = ll_addr[1];
+	l3_addr.s6_addr[10] = ll_addr[2];
+	l3_addr.s6_addr[11] = 0xff;
+	l3_addr.s6_addr[12] = 0xfe;
+	l3_addr.s6_addr[13] = ll_addr[3];
+	l3_addr.s6_addr[14] = ll_addr[4];
+	l3_addr.s6_addr[15] = ll_addr[5];
+
+	char v6addr[50];
+	inet_ntop(AF_INET6, &l3_addr, v6addr, INET6_ADDRSTRLEN);
+	std::string llstr(v6addr);
+    llstr = llstr + "/64";
+
+    ipv6ll = llstr;
+
+    return true;
+}
+
+bool VrrpSync::isAutoGeneratedIPv6LL(const string& vip, const string& ifname)
+{
+    if (vip.compare(0, strlen("fe80"), "fe80"))
+    {
+        return false;
+    }
+
+    /*Check if ll matches with IPv6 LL generated from system mac*/
+    string systemIntf = "eth0";
+    string systemIPv6LL;
+    if (getAutoIPv6LL(systemIntf, systemIPv6LL) == false)
+    {
+        SWSS_LOG_ERROR("Failed to get System mac");
+    }
+
+    /*Check if ll matches with IPv6 LL generated from interface mac
+     * This is necessary since in case of MCLAG, 
+     * Vlan mac changes to Active node System MAC*/
+    string intfIPv6LL;
+    if (getAutoIPv6LL(ifname, intfIPv6LL) == false)
+    {
+        SWSS_LOG_ERROR("Failed to get Interface MAC");
+    }
+
+    SWSS_LOG_NOTICE("VIP: %s Auto System LL %s, Auto Intf LL: %s",
+            vip.c_str(), systemIPv6LL.c_str(), intfIPv6LL.c_str());
+
+	if ((systemIPv6LL == vip) || (intfIPv6LL == vip))
+        return true;
+
+    return false;
+}
+
+
+void VrrpSync::VrrpUpdateNbr(string &ifname, int afi, string &vip, string &op)
+{
+    string cmd, res;
+    int ret;
+    string afi_str = (afi == AF_INET6)? "-6": "";
+
+    cmd = "ip " + afi_str + " neigh flush " + vip + " dev " + ifname;
```

Comment:
I guess it needs to be done only for op = "del".

Reply:
Yes, had written it generically to del/replace. Currently only del is used.